src/lib/utils/Statistic/index.test.js

```diff
@@ -0,0 +1,251 @@
+		expect(statistic.baseUnit).toBe('Wh');
+	});
+
+	it('inverts values in target loads', () => {
+		const data = [
+			{
+				fuel_tech: 'solar_utility',
+				history: {
+					start: '',
+					last: '',
+					interval: '5m',
+					data: [1, 2, -2]
+				}
+			},
+			{
+				fuel_tech: 'battery_discharging',
+				history: {
+					start: '',
+					last: '',
+					interval: '5m',
+					data: [1, 2, -2]
+				}
+			}
+		];
+		let statistic = new Statistic(/** @type {StatsData[]} */ (data), 'history', 'kWh').invertValues(
+			['solar_utility']
+		);
+		expect(statistic.data[0].history.data).toEqual([-1, -2, 2]);
+		expect(statistic.data[1].history.data).toEqual([1, 2, -2]);
+	});
+
+	it('adds total minus loads', () => {
+		const data = [
+			{
+				fuel_tech: 'battery_charging',
+				history: {
+					start: '',
+					last: '',
+					interval: '5m',
+					data: [10, 20, 30]
+				}
+			},
+			{
+				fuel_tech: 'wind',
+				history: {
+					start: '',
+					last: '',
+					interval: '1m',
+					data: [1, 2, 3]
+				}
+			},
+			{
+				fuel_tech: 'solar_utility',
+				history: {
+					start: '',
+					last: '',
+					interval: '1m',
+					data: [2, 4, 6]
+				}
+			}
+		];
+		let statistic = new Statistic(
+			/** @type {StatsData[]} */ (data),
+			'history',
+			'kWh'
+		).addTotalMinusLoads(loadFuelTechs);
+		expect(statistic.data.length).toEqual(statistic.originalData.length + 1);
+		expect(statistic.data[statistic.originalData.length].history.data).toEqual([3, 6, 9]);
+	});
+
+	it('merges and interpolates', () => {
+		const data = [
+			{
+				fuel_tech: 'battery_charging',
+				history: {
+					start: '2024-07-17T09:35:00+00:00',
+					last: '2024-07-17T09:36:00+00:00',
+					interval: '1m',
+					data: [10, 20]
+				},
+				forecast: {
+					start: '2024-07-17T09:37:00+00:00',
+					last: '2024-07-17T09:38:00+00:00',
+					interval: '1m',
+					data: [40, 50]
+				}
+			},
+			{
+				fuel_tech: 'wind',
+				history: {
+					start: '2024-07-17T09:25:00+00:00',
+					last: '2024-07-17T09:35:00+00:00',
+					interval: '5m',
+					data: [5, 4, 3]
+				},
+				forecast: {
+					start: '2024-07-17T12:35:00+00:00',
+					last: '2024-07-17T12:45:00+00:00',
+					interval: '10m',
+					data: [1, 2, 3]
+				}
+			}
+		];
+		let statistic = new Statistic(
+			/** @type {StatsData[]} */ (data),
+			'history',
+			'kWh'
+		).mergeAndInterpolate();
+		expect(statistic.data[0].forecast).toEqual(data[0].forecast);
+		expect(statistic.data[0].history).not.toEqual(data[0].history);
+		expect(statistic.data[0].history.data).toEqual([
+			...data[0].history.data,
+			...data[0].forecast.data
+		]);
+		expect(statistic.data[0].history.start).toEqual(data[0].history.start);
+		expect(statistic.data[0].history.last).toEqual(data[0].forecast.last);
+		expect(statistic.data[1].forecast).toEqual(data[1].forecast);
+		expect(statistic.data[1].history).not.toEqual(data[1].history);
+		// TODO: don't fully understand the intended behaviour here
```

Comment:
See here re intended behaviour. Why 4 values? Why does last get set to 09:38?

Reply:
not entirely sure, but I am guessing this is trying to test a 1min interval interpolation.

Reply:
@chienleng yes - I wrote the test, in case that wasn't clear :D This was just related to not fully understanding the intent behind the behaviour around start/last and the number of interpolated values - and therefore harder to write a meaningful test for it. So long as you think the test actually makes sense, then 👍

Reply:
ah, got it. :) Understood about the confusion of those values. The Stats class will need a refactor at some point, I am mostly porting this from a previous code base from OpenNEM. 

I will take a closer look at reviewing this some time this week. 

thanks for your help though.